taskmgr/queueservice_test.go

```diff
@@ -15,49 +19,736 @@ import (
+		w.Write([]byte(message))
+	}
+
+	s := httptest.NewServer(http.HandlerFunc(handler))
+	defer s.Close()
+	logger, _ := runtime.CreateLogger(os.Getenv("LOGGING_LEVEL"))
+	service := queueService{
+		Log:              logger.WithField("component", "Queue Service"),
+		ExpirationOffset: 300,
+		Expires:          tcclient.Time(time.Now().Add(time.Minute * 10)),
+		queues: []taskQueue{
+			{
+				SignedDeleteUrl: fmt.Sprintf("%s/delete/{{messageId}}/{{popReceipt}}", s.URL),
+				SignedPollUrl:   fmt.Sprintf("%s/tasks/1234?messages=true", s.URL),
+			},
+			{
+				SignedDeleteUrl: fmt.Sprintf("%s/delete/{{messageId}}/{{popReceipt}}", s.URL),
+				SignedPollUrl:   fmt.Sprintf("%s/tasks/456?messages=true", s.URL),
+			},
+		},
+	}
+
+	tasks, err := service.retrieveTasksFromQueue(2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Only two tasks should have been retrieving leaving the third task in the lower
+	// priority queue not being retrieved
+	assert.Equal(t, 2, len(tasks))
+	taskIds := []string{"def", "ghi"}
+	for i, v := range taskIds {
+		assert.Equal(t, v, tasks[i].TaskId)
+	}
+}
+func TestRetrieveTasksFromQueueDequeueChecked(t *testing.T) {
+	// Dequeue count is above the threshold of 15
+	message := `<?xml version="1.0" encoding="utf-8"?>
+	<QueueMessagesList>
+	  <QueueMessage>
+		<MessageId>5974b586-0df3-4e2d-ad0c-18e3892bfca3</MessageId>
+		<InsertionTime>Fri, 09 Oct 2009 21:04:30 GMT</InsertionTime>
+		<ExpirationTime>Fri, 16 Oct 2009 21:04:30 GMT</ExpirationTime>
+		<PopReceipt>YzQ4Yzg1MDItYTc0Ny00OWNjLTkxYTUtZGM0MDFiZDAwYzEw</PopReceipt>
+		<TimeNextVisible>Fri, 09 Oct 2009 23:29:20 GMT</TimeNextVisible>
+		<DequeueCount>16</DequeueCount>
+		<MessageText>eyJ0YXNrSWQiOiAiYWJjIiwgInJ1bklkIjogMX0=</MessageText>
+	  </QueueMessage>
+	</QueueMessagesList>`
+
+	// Delete URL should be called when dequeue count above threshold (15)
+	deleteCalled := false
+
+	var handler = func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/delete/5974b586-0df3-4e2d-ad0c-18e3892bfca3/YzQ4Yzg1MDItYTc0Ny00OWNjLTkxYTUtZGM0MDFiZDAwYzEw" {
+			deleteCalled = true
+			return
+		}
+		w.Header().Set("Content-Type", "application/xml")
+		w.Write([]byte(message))
+	}
+
+	s := httptest.NewServer(http.HandlerFunc(handler))
+	defer s.Close()
+	logger, _ := runtime.CreateLogger(os.Getenv("LOGGING_LEVEL"))
+	service := queueService{
+		Log:              logger.WithField("component", "Queue Service"),
+		ExpirationOffset: 300,
+		Expires:          tcclient.Time(time.Now().Add(time.Minute * 10)),
+		queues: []taskQueue{
+			{
+				SignedDeleteUrl: fmt.Sprintf("%s/delete/{{messageId}}/{{popReceipt}}", s.URL),
+				SignedPollUrl:   fmt.Sprintf("%s/tasks/1234?messages=true", s.URL),
+			},
+		},
+	}
+
+	tasks, err := service.retrieveTasksFromQueue(2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, 1, len(tasks))
+	assert.True(t, deleteCalled, "Delete should have been called when dequeue count above threshold")
+
+}
+
+func TestClaimTask(t *testing.T) {
```

Comment:
Some of these functions are quite long, so it might be helpful to add a sentence or two to the function comments to give a high level about what the test is checking.